### ISSUE TITLE
🏛 Proposal: RETAB EVERYTHING!

### ISSUE THREAD

**Issue by pschmitt** [zdharma-continuum/zinit#115]
The more I work with zinit's core the more my eyes bleed.
Some lines are absurdly long.
A few examples:

- https://github.com/zdharma-continuum/zinit/blob/8b5ed2a59768b5792853ae162a8e0a7eb4594f1c/zinit-install.zsh#L4
- https://github.com/zdharma-continuum/zinit/blob/8b5ed2a59768b5792853ae162a8e0a7eb4594f1c/zinit-install.zsh#L995
- https://github.com/zdharma-continuum/zinit/blob/8b5ed2a59768b5792853ae162a8e0a7eb4594f1c/zinit-install.zsh#L1848

I get that most people don't stick to 80 chars for their code but I'd at least appreciate if we re-tabbed the zsh scripts to use 2 space chars instead of 4.

**TL;DR**
I'm advocating for `vim: ft=zsh et ts=2 sw=2` for `zinit*.zsh`

**Comment by alichtman** [comment]
I agree that we have a readability problem to solve here. Personally, I like ts/sw=4, but I see the argument for using ts=2 here. 

That said, I think the real fix is breaking up some of the massive one-liners. For example, the first link you included has 0 leading spaces but is still unruly to read and work with.

**Comment by pschmitt** [comment]
Some lines we cannot split easily without refactoring it all and possibly (probably) hurting performance. zsh expansion is powerful, but afaik there's no readable way to split these lines other than storing temporary results in a variable and doing the rest of the work in a second statement.

The best way forward here is to add a bunch of comments explaining the weirdest exps IMO. And unspaghetti what we encounter along the way. For instance I "fixed" the first example in #116:

https://github.com/zdharma-continuum/zinit/blob/7439160162ce0d60a0bdba02447baf23caf2ec66/zinit-install.zsh#L4-L7

Doing it all in one go would be a huge undertaking.

Another thing: to make things better I'm all for adopting a "return early" approach (I don't know the real tech term) ie:

Instead of:

```
fun() {
	  if check_is_okay {
	    long_block_of_code
	    return 0
	  } elif check_not_okay {
	    another_block_of_code
	    return 1
	  }
}
```

that:

```
fun() {
	  if check_not_okay {
	    another_block_of_code
	    return 1
	  }
	  # check is okay here
	  block_of_code
	  return 0
}
```

It's a poor example, but I'm sure you get my point.

@alichtman I fixed the

**Comment by alichtman** [comment]
> Doing it all in one go

Yep, minor fixes over a long period of time is alright. Any improvements > no improvements :)

**Comment by vladdoster** [comment]
Closing in favor of #238.
